blob/oc_test.go

```diff
@@ -45,20 +51,61 @@ func TestOpenCensus(t *testing.T) {
 	if _, err := b.ReadAll(ctx, "noSuchKey"); err == nil {
 		t.Fatal("got nil, want error")
 	}
-	// TODO(jba): after #1213, use DiffSpans.
 
-	data := <-te.Stats
-	for i, r := range data.Rows {
-		fmt.Printf("%d: %s %T\n", i, r, r.Data)
+	diff := octest.DiffSpans(te.Spans, []octest.Span{
+		{"gocloud.dev/blob.NewWriter", gcerrors.OK},
+		{"gocloud.dev/blob.NewRangeReader", gcerrors.OK},
+		{"gocloud.dev/blob.Attributes", gcerrors.OK},
+		{"gocloud.dev/blob.Delete", gcerrors.OK},
+		{"gocloud.dev/blob.NewRangeReader", gcerrors.NotFound},
+	})
+	if diff != "" {
+		t.Errorf("trace: %s", diff)
 	}
 
-	data = <-te.Stats
-	for i, r := range data.Rows {
-		fmt.Printf("%d: %s %T\n", i, r, r.Data)
+	// Wait for counts. Expect all counts to arrive in the same view.Data.
+	for {
+		data := <-te.Stats
+		if _, ok := data.Rows[0].Data.(*view.CountData); !ok {
+			continue
+		}
+		// There should be a count of 1 for each of these sets of tags.
+		// Rows may appear in any order, but tags within a row will be sorted
+		// by key name (see https://godoc.org/go.opencensus.io/stats/view#Row.Equal).
+		var want []*view.Row
+		for _, m := range []string{"Attributes", "Delete", "NewRangeReader", "NewWriter"} {
+			want = append(want, &view.Row{
+				Tags: []tag.Tag{
+					{Key: oc.MethodKey, Value: "gocloud.dev/blob." + m},
+					{Key: oc.ProviderKey, Value: "gocloud.dev/blob/memblob"},
+					{Key: oc.StatusKey, Value: "OK"},
+				},
+				Data: &view.CountData{Value: 1},
+			})
+		}
+		want = append(want, &view.Row{
+			Tags: []tag.Tag{
+				{Key: oc.MethodKey, Value: "gocloud.dev/blob.NewRangeReader"},
+				{Key: oc.ProviderKey, Value: "gocloud.dev/blob/memblob"},
+				{Key: oc.StatusKey, Value: "NotFound"},
+			},
+			Data: &view.CountData{Value: 1},
+		})
+		diff := cmp.Diff(data.Rows, want, cmpopts.SortSlices(lessRow))
+		if diff != "" {
+			t.Errorf("metrics: %s", diff)
+		}
+		break
 	}
+}
+
+// The order doesn't matter; we just need a deterministic ordering of rows.
+func lessRow(r1, r2 *view.Row) bool { return rowKey(r1) < rowKey(r2) }
```

Comment:
Shouldn't this be in `oc` somewhere? Seems like you'll need it for every test.

Reply:
Yes, I'm planning on factoring out the testing code  at some point. Probably with my next metrics integration.